### ISSUE TITLE
Configuration Architecture Issue

### ISSUE THREAD

**Issue by J4D3N** [J4D3N/GameRanks#22]
The configuration needs to be redone, the problem with the current version is there can be multiple ranks with the same name but with different case. This can be a problem when trying to get the rank based on a string because we cannot tell if the request is case sensitive or not.

I'm proposing a change to a new configuration file type preferably JSON. That way we can make rank names more display customizable and more compatible.

**Comment by Tsuser1** [comment]
I agree, I will move to the JSON config architecture once implemented.

**Comment by J4D3N** [comment]
How do we go about converting old configuration to the new?

**Comment by Tsuser1** [comment]
`plugin.getConfig().getConfigurationSection(true);` Loop through all values and copy them over to JSON.

**Comment by Tsuser1** [comment]
I think a "1.1.0.5" update should be pushed to Spigot so known bugs are fixed, also the pager is a wanted feature.

**Comment by J4D3N** [comment]
We can implement the permissions feature before we push a beta release for 1.1.1?

**Comment by Tsuser1** [comment]
Yeah, that would be best.

**Comment by J4D3N** [comment]
I'm working on the permissions now but how will handle deranking? Do you need permission to derank and should this permissions system be optional?

**Comment by Tsuser1** [comment]
Permissions system can be overridden with `gameranks.rankup.*`

**Comment by J4D3N** [comment]
Technically what if the admin wanted players to derank but only a specified group?

**Comment by Tsuser1** [comment]
Will implement.

**Comment by J4D3N** [comment]
I just hooked your IF statement to the config value of whether to use permissions or not.

**Comment by Tsuser1** [comment]
Okay

**Comment by J4D3N** [comment]
I have a feeling this problem here is going to be an annoyance to fix.

**Comment by Tsuser1** [comment]
This might be one for 1.1.3, its going to be a huge issue.

**Comment by J4D3N** [comment]
I'm going to focus on this little configuration rewrite, and hopefully will have some new configuration and language utility in the upcoming days to ease the message responses.

**Comment by J4D3N** [comment]
I should be able to port the class to a JSON file without having to change too much as usage goes.

**Comment by Tsuser1** [comment]
Yeah, its probably better that this is fixed now rather than later.

**Comment by J4D3N** [comment]
You got Skype?

**Comment by Tsuser1** [comment]
Yes, `tsuserpbx`
